Scripts/Game/UI/Context/OVT_RecruitsContext.c

```diff
@@ -0,0 +1,474 @@
+class OVT_RecruitsContext : OVT_UIContext
+{
+	[Attribute("{5D5C558A6E391690}UI/Layouts/Menu/RecruitsMenu/RecruitListItem.layout", uiwidget: UIWidgets.ResourceNamePicker, desc: "Layout for recruit list items", params: "layout")]
+	ResourceName m_RecruitItemLayout;
+	
+	protected OVT_RecruitManagerComponent m_RecruitManager;
+	protected ref array<ref OVT_RecruitData> m_aPlayerRecruits;
+	protected OVT_RecruitData m_SelectedRecruit;
+	protected Widget m_wSelectedWidget;
+	protected int m_iSelectedIndex = -1;
+	protected OVT_RecruitData m_CurrentRenamingRecruit;
+	protected SCR_ConfigurableDialogUi m_RenameDialog;
+	
+	override void PostInit()
+	{		
+		m_RecruitManager = OVT_RecruitManagerComponent.GetInstance();
+	}
+	
+	override void OnShow()
+	{			
+		// Set up button click handlers
+		Widget showOnMapButton = m_wRoot.FindAnyWidget("ShowOnMapButton");
+		if (showOnMapButton)
+		{
+			SCR_InputButtonComponent action = SCR_InputButtonComponent.Cast(showOnMapButton.FindHandler(SCR_InputButtonComponent));
+			if (action)
+				action.m_OnActivated.Insert(ShowOnMap);
+		}
+		
+		Widget renameButton = m_wRoot.FindAnyWidget("RenameButton");
+		if (renameButton)
+		{
+			SCR_InputButtonComponent action = SCR_InputButtonComponent.Cast(renameButton.FindHandler(SCR_InputButtonComponent));
+			if (action)
+				action.m_OnActivated.Insert(RenameRecruit);
+		}
+		
+		Widget dismissButton = m_wRoot.FindAnyWidget("DismissButton");
+		if (dismissButton)
+		{
+			SCR_InputButtonComponent action = SCR_InputButtonComponent.Cast(dismissButton.FindHandler(SCR_InputButtonComponent));
+			if (action)
+				action.m_OnActivated.Insert(DismissRecruit);
+		}
+		
+		Refresh();
+	}
+	
+	override void OnFrame(float timeSlice)
+	{
+		super.OnFrame(timeSlice);
+		
+		// Handle recruit list navigation
+		if (m_InputManager.GetActionTriggered("MenuUp"))
```

Comment:
These actions fire when any menu is open (Overthrow menu, Game Master etc), resulting in VME's